src/frontends/ir/src/os/win/win_mmap_object.cpp

```diff
@@ -58,13 +58,13 @@ class MapHolder {
     }
 
     void set(const std::string& path) {
-        auto h = ::CreateFileA(path.c_str(), GENERIC_READ, FILE_SHARE_READ, 0, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, 0);
+        auto h = ::CreateFileA(path.c_str(), GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_DELETE, 0, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, 0);
         map(path, h);
     }
 
 #ifdef OPENVINO_ENABLE_UNICODE_PATH_SUPPORT
     void set(const std::wstring& path) {
-        auto h = ::CreateFileW(path.c_str(), GENERIC_READ, FILE_SHARE_READ, 0, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, 0);
+        auto h = ::CreateFileW(path.c_str(), GENERIC_READ, FILE_SHARE_READ | FILE_SHARE_DELETE, 0, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, 0);
```

Comment:
why do we need this new flag?

Reply:
Linux allows rename/deletion of files even if not all handles on file are closed (default behavior). To achieve ~same behavior on Windows, need to add `FILE_SHARE_DELETE` key (or deletion will fail like `File is used by <name>.exe`)

Reply:
`FILE_SHARE_DELETE` key will be removed because it's platform specific (`NTFS` allows to delete file if there are open handles on file, but `FAT32` - not), and we have no specific requirements to support it. If someone really need it, he/she may switch `mmap` to `read`